_posts/2019-01-30-year-of-the-dog.md

```diff
@@ -0,0 +1,199 @@
+categories:
+- Stack
+- GraphQL
+- Typescript
+- API
+title: What changed in our stack in 2018
+author_staff_member: 03_tom
+---
+
+As our team and codebase grows, we sometimes find the need to cast a look back
+at how things changed, and remember fondly the code we wrote and deleted, the
+abstractions we built and scrapped, how our code used to be, and how much fun
+it was to get it to where it is now.
+
+We are building a platform to make live marketing scalable, measurable and
+targeted — as such we have multiple apps for the various actors of the platform
+and one API.  We want to make sure we can keep growing and adapting the
+platform to changing needs, maintaining agility and stability in the process.
+This is our high-level mindset when making decisions regarding our tech stack.
+This post is a recap of how it has worked for us in 2018.
+
+![store2be developer, early 2018]({{ site.url }}/images/2018_retrospective/grumpy_pallas_cat_c_tambako_the_jaguar_flickr.jpg)
+_store2be developer, early 2018 (photo © Tambako The Jaguar, flickr)_
+
+## Frontend - the React apps
+
+We introduced Typescript in our React apps at the end of 2017, and we are only
+now getting close to completely typed frontend apps. Before Typescript, we had
+suffered from the churn in our JS architecture as we experimented with more
+scaleable approaches to state management and asynchronous actions - we had our
+own abstractions on top of
+[redux-saga](https://github.com/redux-saga/redux-saga) and even open sourced
+our way to deal with boilerplate in redux,
+[redux-belt](https://github.com/store2be/redux-belt/). These abstractions were
+constantly evolving, and it was a challenge to evolve core abstractions in very
+stateful apps without introducing type errors which translated directly to
+crashes for our users. There are enough Typescript success stories on the
+internet, so I will not elaborate too much on this.
+
+Good tooling is something we appreciate, so in addition to Typescript we
+adopted [prettier](https://prettier.io/),
+[TSLint](https://github.com/palantir/tslint/) (in addition to ESLint),
+[jest](https://jestjs.io/) (replacing jasmine), and [snapshot
+testing](https://jestjs.io/docs/en/snapshot-testing.html). All of these lifted
+a lot of mental burden: no more worrying about or manually fiddling with code
+formatting, automatically enforced conventions, more certainty that the UI does
+not change by surprise, etc. Linting, testing and a lot more is enforced in CI so
+we can focus on the hard problems during code reviews.
+
+We liked the experience so much we implemented a backend service for email templating
+in typescript and [wrote about
+it](https://tech.store2be.com/email/sendwithus/mjml/typescript/react/2018/06/14/email-templates-at-store2be-and-gdpr/).
+
+Another notable and much appreciated addition to our toolbox was
+[react-storybook](https://github.com/storybooks/storybook) for our shared UI
+components.
+
+## An unexpected journey - GraphQL
```

Comment:
We could change this to `An unexpected journey - GraphQL in the frontend` to make it clear, that you talk about the backend later.